modules/eks/main.tf

```diff
@@ -2,7 +2,7 @@ locals {
   # Map this module config to the upstream module config
   eks_node_group_config = { for n, config in var.eks_node_groups :
     n => {
-      name = "${var.cluster_name}-${n}"
+      name_prefix = "${var.cluster_name}-${n}"
```

Comment:
How about a variable to let people use either `name` or `name_prefix` for backward compatibility? Otherwise when people use this version they will need to recreate their node groups.

Reply:
Agree, I added a new variable `node_group_name_as_prefix` it for backwards compatibility and set default to `false`. So default to the old behaviour.